test/common/common/utility_test.cc

```diff
@@ -1095,6 +1095,60 @@ TEST(InlineStorageTest, InlineString) {
   EXPECT_EQ("Hello, world!", hello->toString());
 }
 
+TEST(TinyStringTest, Empty) {
```

Comment:
If we think this approach makes sense and it is worth it, unit-tests for NumericEncoding need to be added to this PR.